src/NumericalAlgorithms/Spectral/Spectral.cpp

```diff
@@ -152,6 +152,30 @@ struct DifferentiationMatrixGenerator {
   }
 };
 
+template <Basis BasisType, Quadrature QuadratureType>
+struct WeakFluxDifferentiationMatrixGenerator {
+  Matrix operator()(const size_t num_points) const noexcept {
+    if (BasisType != Basis::Legendre) {
+      ERROR(
+          "Weak differentiation matrix only implemented for Legendre "
+          "polynomials.");
+    }
+    const DataVector& weights =
+        quadrature_weights<BasisType, QuadratureType>(num_points);
+
+    Matrix weak_diff_matrix =
+        DifferentiationMatrixGenerator<BasisType, QuadratureType>{}.operator()(
+            num_points);
+    transpose(weak_diff_matrix);
+    for (size_t i = 0; i < weak_diff_matrix.rows(); ++i) {
+      for (size_t j = 0; j < weak_diff_matrix.rows(); ++j) {
```

Comment:
Why not use the variable `num_points` to set the upper bound here?

And if you prefer to make no assumptions about the matrix dimensions, then one of these rows should probably be a columns?

Reply:
Switched to `num_points`, the matrix always needs to be square anyway :P